logger_bench_test.go

```diff
@@ -238,3 +240,132 @@ func Benchmark100Fields(b *testing.B) {
 		logger.With(first...).Info("Child loggers with lots of context.", second...)
 	}
 }
+
+func dummy(wg *sync.WaitGroup, s string, i int) string {
+	if i == 0 {
+		wg.Wait()
+		return "1" + s
+	}
+	return dummy(wg, s, i-1)
+}
+
+// increaseAvgStack starts a background goroutine with a variable
+// stack size. The goal is to move the average stack size higher,
+// since https://go-review.googlesource.com/c/go/+/345889 this affects
+// goroutine starting stack size.
+func increaseAvgStack(n int) *sync.WaitGroup {
+	wg := sync.WaitGroup{}
+	wg.Add(1)
+
+	go dummy(&wg, "hi", n)
+	return &wg
+}
+
+func BenchmarkAny(b *testing.B) {
+	key := "some-long-string-longer-than-16"
+
+	tests := []struct {
+		name   string
+		typed  func() Field
+		anyArg any
+	}{
+		{
+			name:   "string",
+			typed:  func() Field { return String(key, "yet-another-long-string") },
+			anyArg: "yet-another-long-string",
+		},
+		{
+			name:   "stringer",
+			typed:  func() Field { return Stringer(key, InfoLevel) },
+			anyArg: InfoLevel,
+		},
+	}
+
+	for _, tt := range tests {
+		b.Run(tt.name, func(b *testing.B) {
+			b.Run("field-only", func(b *testing.B) {
+				b.Run("typed", func(b *testing.B) {
+					withBenchedLogger(b, func(log *Logger) {
+						f := tt.typed()
+						runtime.KeepAlive(f)
+					})
+				})
+				b.Run("any", func(b *testing.B) {
+					withBenchedLogger(b, func(log *Logger) {
+						f := Any(key, tt.anyArg)
+						runtime.KeepAlive(f)
+					})
+				})
+			})
+			b.Run("log", func(b *testing.B) {
+				b.Run("typed", func(b *testing.B) {
+					withBenchedLogger(b, func(log *Logger) {
+						log.Info("", tt.typed())
+					})
+				})
+				b.Run("any", func(b *testing.B) {
+					withBenchedLogger(b, func(log *Logger) {
+						log.Info("", Any(key, tt.anyArg))
+					})
+				})
+			})
+			b.Run("log-go", func(b *testing.B) {
+				b.Run("typed", func(b *testing.B) {
+					wg := sync.WaitGroup{}
+					wg.Add(b.N)
+					b.ResetTimer()
+					withBenchedLogger(b, func(log *Logger) {
+						go func() {
+							log.Info("", tt.typed())
+							wg.Done()
+						}()
+					})
+					wg.Wait()
+				})
+				b.Run("any", func(b *testing.B) {
+					wg := sync.WaitGroup{}
+					wg.Add(b.N)
+					b.ResetTimer()
+					withBenchedLogger(b, func(log *Logger) {
+						go func() {
+							log.Info("", Any(key, tt.anyArg))
+							wg.Done()
+						}()
+					})
+					wg.Wait()
```

Comment:
Are you sure this works? (Related to the prior comment about `b.N` goroutines.)
Each iteration (whether with RunParallel or with a plain for loop), will last only as long as spawning the goroutine takes. I'm not sure this will do what we want? 
I know we're trying to measure the cost inside a new goroutine, but this doesn't seem right?
Maybe we need to spawn and wait for the goroutine in each iteration of the loop?

```go
var wg sync.WaitGroup
wg.Add(1)
go func() {
    ...
    wg.Done()
}()
wg.Wait()
```

This will measure the cost of the whole operation (including spawning the goroutine) per iteration.

Reply:
Hm.

It was testing something reasonable, because the changes we were making to the stack were clearly having an effect. I agree that this is better though, thanks.